SC-419: How to for RBAC

harrycguo opened the pull request:
### Your checklist for this pull request

Thank you for your contribution to the `next-dev-docs` repo. 

🚨Please review the [Pull request guidelines](../contributor-guide/contributor-guide.md#pull-request-guidelines) to this repository before submitting this PR.

- [ ] Your code builds clean without any errors or warnings
  
  ```bash
  yarn build
  ```

- [ ] You are using approved terminology

  Refer to the [style guide](../contributor-guide/style-guide.md) for help.


### Description 

JIRA Story: https://beyondidentity.atlassian.net/browse/SC-419

I moved some things around, notably:

To top level "How to Guides":

- Create a role
- Add users and groups to a role
- Configure RBAC

To the "Administrative" section:
- Configure Administrative Roles

Will5 commented on docs/how-to/configure-rbac.mdx:
```diff
@@ -1,29 +1,79 @@
+Resource servers represent an entity hosting protected resources that the client is attempting to access. This is the entire set of permissions that are allowed on a resource.
+These permissions are expressed through scopes, which are strings that represent the set of permissions.
+
+For our example above, we would create a resource server called Shoes and the scopes that are available could be `shoes:read`, `shoes:buy`, `clearance:buy`, `new-arrivals:read`, `new-arrivals:buy`, etc.
```
We should probably link to the "Add a resource server" doc.

turnipdabeets commented on docs/how-to/configure-rbac.mdx:
```diff
@@ -1,29 +1,79 @@
+
+## Token Introspection
+
+Whenever a [token introspection](<link to token introspection>) is called for an access token with an identity as the principal, the scopes returned in the introspect
```
need link here :)

turnipdabeets commented on docs/how-to/configure-rbac.mdx:
```diff
@@ -1,29 +1,79 @@
+
+## Token Introspection
+
+Whenever a [token introspection](<link to token introspection>) is called for an access token with an identity as the principal, the scopes returned in the introspect
+response will contain **only a subset of the scopes for which the identity has the role(s) for**.
```
```suggestion
When invoking [token introspection](<link to token introspection>) for an access token associated with an identity as the principal, the returned scopes in the introspection response will contain **only a subset of the scopes for which the identity has the role(s) for**.
```

turnipdabeets commented on docs/how-to/configure-rbac.mdx:
```diff
@@ -1,29 +1,79 @@
+
+## Putting it all together
+
+After creating the aforementioned resource server and roles above, we would create an application with the authorization code flow and integrate that with your application.
+Now, whenever a user logs in and attempts to access the various parts of your website, RBAC will ensure that the logged in user can only access the resources that are allowed
+to them by their role.
+
+RBAC is an authorization tool that can be used to validate access permissions on the end user, and is just one of the tools that Beyond Identity can help you accomplish your
+identity and access management goals.
```
```suggestion
After creating the aforementioned resource server and roles above, the next step involves the creation of an application employing the authorization code flow using the roles and scopes created. Consequently, whenever a user logs in and attempts to access different sections of your website, Role-Based Access Control (RBAC) will ensure that the logged in user can only access the resources that are allowed to them by their designated role.

RBAC is an authorization tool that can be used to validate access permissions on the end user, and is just one of the tools that Beyond Identity can help you accomplish your identity and access management goals.
```

turnipdabeets commented on docs/how-to/configure-rbac.mdx:
```diff
@@ -1,29 +1,79 @@
+Now, whenever a user logs in and attempts to access the various parts of your website, RBAC will ensure that the logged in user can only access the resources that are allowed
+to them by their role.
+
+RBAC is an authorization tool that can be used to validate access permissions on the end user, and is just one of the tools that Beyond Identity can help you accomplish your
```
Is there a logical next place to link a user for more tools to accomplish identity and access management goals?

harrycguo replied:
I don't think so, this is it!

turnipdabeets commented on docs/how-to/configure-rbac.mdx:
```diff
@@ -1,29 +1,79 @@
+
+## Putting it all together
+
+After creating the aforementioned resource server and roles above, we would create an application with the authorization code flow and integrate that with your application.
```
Is there anything we should point out in how to integrate that with the application?

harrycguo replied:
Hmmm I'm not sure if there is anything besides this step. The developer must take steps to introspect within their own application after this.

turnipdabeets reviewed (commented):
Looks good! A couple of rewordings and a few questions.

harrycguo commented:
@Will5 @turnipdabeets I added this to the PR description, but I moved some things around, notably:

To top level "How to Guides":

- Create a role
- Add users and groups to a role
- Configure RBAC

To the "Administrative" section I added:
- Configure Administrative Roles (basic page on configuring admin roles)
